### PR TITLE
k8s support with instructions

### PR DESCRIPTION
**Pull request by cleverguy25**
Adds basic instructions and setup for deploying in k8s.  

Right now we are not publishing an official docker container until we see if there is demand/need, the docker file is easy to build and use.

Fixes [#17242](https://github.com/pulumi/pulumi-service/issues/17242)
Fixes https://github.com/pulumi/customer-managed-deployment-agent/issues/1

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN apk add curl 
+
+ENV HOME /home/builder
+
+RUN curl -fsSL https://raw.githubusercontent.com/pulumi/customer-managed-deployment-agent/main/install.sh | sh
+
+FROM alpine:3.18
+
+# RUN apk add curl && curl -fsSL https://raw.githubusercontent.com/pulumi/customer-managed-deployment-agent/main/install.sh | sh
```
Comment (dschaller) [inline docker/Dockerfile:11]:
```suggestion
```

File: kubernetes/Pulumi.dev.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+config:
+  pulumi-agent-kubernetes:agentImage: pulumi/pulumi-agent:curl
+  pulumi-agent-kubernetes:agentNamespace: cleve
```
Comment (dschaller) [inline kubernetes/Pulumi.dev.yaml:3]: Should we make the dev namespace more generic

File: kubernetes/agent.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                            app: "pulumi-deployment-agent-pool",
+                            "app.kubernetes.io/name": "pulumi-deployment-agent-pool",
+                        },
+                    },
+                    spec: {
+                        containers: [
+                        {
+                            name: "agent",
+                            image: args.imageName,
+                            imagePullPolicy: "IfNotPresent",
```
Comment (dschaller) [inline kubernetes/agent.ts:52]: Do we want if not present over always? If you're using a tag that doesn't change (e.g. `latest`) and the image with this tag is already on the node you'll never pull a newer version (even if one has been published)

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM --platform=$BUILDPLATFORM alpine:3.18 as buildenv
+
+RUN apk add curl 
+
+ENV HOME /home/builder
+
+RUN curl -fsSL https://raw.githubusercontent.com/pulumi/customer-managed-deployment-agent/main/install.sh | sh
+
+FROM alpine:3.18
```
Comment (lblackstone) [inline docker/Dockerfile:9]: Do you need alpine, or would the scratch image work?

File: kubernetes/Pulumi.dev.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+  pulumi-agent-kubernetes:selfHostedAgentsAccessToken:
```
Comment (lblackstone) [inline kubernetes/Pulumi.dev.yaml:4]: Actually, I wonder if this file should be committed at all?

File: kubernetes/Pulumi.yaml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+name: pulumi-agent-kubernetes
+runtime: nodejs
+description: Installs customer managed kubernetes agent in a Kubernetes cluster
```
Comment (lblackstone) [inline kubernetes/Pulumi.yaml:3]:
```suggestion
description: Installs customer managed deployment agent in a Kubernetes cluster
```

**Review by lblackstone — APPROVED** [review]
Just a few minor comments. I'm ok to merge if you need to unblock further testing.

**Review by dschaller — APPROVED** [review]
Looks good with a few minor questions

File: docker/Dockerfile (hunk 4 above)
Comment (cleverguy25) [reply docker/Dockerfile:9]: I believe I need alpine, but cannot remember why, but have not tested. For now I would like to just go with this and can simplify later if there is demand.

File: kubernetes/agent.ts (hunk 3 above)
Comment (cleverguy25) [reply kubernetes/agent.ts:52]: I will make configurable, as it depends on the imageName and what people want from their setup

**Pull request merged**